ui/component/or-dashboard-builder/src/widgets/or-image-widget.ts

```diff
@@ -0,0 +1,366 @@
+            });
+        }
+    }
+
+    // Fetching the assets according to the AttributeRef[] input in DashboardWidget if required. TODO: Simplify this to only request data needed for attribute list
+    async fetchAssets(config: OrWidgetConfig | any): Promise<Asset[] | undefined> {
+        if (config.attributeRefs && config.attributeRefs.length > 0) {
+            let assets: Asset[] = [];
+            await manager.rest.api.AssetResource.queryAssets({
+                ids: config.attributeRefs?.map((x: AttributeRef) => x.id) as string[],
+                realm: {
+                    name: this.realm
+                },
+                select: {
+                    attributes: config.attributeRefs?.map((x: AttributeRef) => x.name) as string[]
+                }
+            }).then(response => {
+                assets = response.data;
+            }).catch((reason) => {
+                console.error(reason);
+                showSnackbar(undefined, i18next.t('errorOccurred'));
+            });
+            return assets;
+        }
+    }
+}
+
+@customElement("or-image-widgetsettings")
+export class OrImageWidgetSettings extends LitElement {
+    static get styles() {
+        return [style, widgetSettingsStyling];
+    }
+
+    private _fileElem!: HTMLInputElement;
+    private expandedPanels: string[] = [i18next.t('attributes'), i18next.t('marker coordinates'), i18next.t('image settings')];
+    private loadedAssets?: Asset[];
+    @property()
+    public readonly widget?: DashboardWidget;
+
+    // same implementation as /or-components/src/or-file-uploader.ts
+    // file explorer dialogue opens with supported file types
+    @property({ attribute: false })
+    public accept: string = "image/png,image/jpeg,image/vnd.microsoft.icon,image/svg+xml";
+    // UI Rendering
+    render() {
+        const config = JSON.parse(JSON.stringify(this.widget!.widgetConfig)) as ImageWidgetConfig; // duplicate to edit, to prevent parent updates. Please trigger updateConfig()
+
+        var output = html`
+            <div>
+                ${this.generateExpandableHeader(i18next.t('attributes'))}
+            </div>
+            <div>
+                ${this.expandedPanels.includes(i18next.t('attributes')) ? html`
+                    <or-dashboard-settingspanel .type="${SettingsPanelType.MULTI_ATTRIBUTE}" .widgetConfig="${this.widget!.widgetConfig}"
+                    @updated="${(event: CustomEvent) => {
+                        this.onAttributesUpdate(event.detail.changes);
+                        this.updateConfig(this.widget!, event.detail.changes.get('config'));
+                }}"
+                    ></or-dashboard-settingspanel>
+                ` : null}
+        </div>
+        <div>
+            ${this.generateExpandableHeader(i18next.t('marker coordinates'))}
+        </div>
+        <div style="display: flex; justify-content: start; flex-direction: row; flex-wrap: wrap; align-items: flex-start; min-width: 150px;">
```

Comment:
I'm also guilty for doing a lot of inline CSS, so you're not the only one haha
It would be better if it's in a seperate `style.ts` file of course.
Make it part of the code cleanup I guess 😄